### PR TITLE
add-nodeSelector

### PR DESCRIPTION
**Pull request by magorKathy**
**What type of PR is this?**

**What this PR does / why we need it**:
It's needed so we can configure a NodeSelector for Envoy gateway  pods within values file.
A common use case for nodeSelector configuration is to control the placement of pods on specific nodes within a cluster based on node labels. This allows to ensure that certain workloads run only on nodes that meet specific criteria, such as hardware specifications, geographic location, or other custom attributes.

**Which issue(s) this PR fixes**:
N/A

### CODE REVIEW

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -56,6 +56,7 @@ deployment:
     labels: {}
     topologySpreadConstraints: []
     tolerations: []
+  nodeSelector: {}
```
Comment (arkodg) [inline charts/gateway-helm/values.tmpl.yaml:59]: should live at the same level/identation as `topologySpreadConstraints`

**Comment by arkodg** [comment]
thanks for raising this PR @magorKathy, added a comment

File: charts/gateway-helm/values.tmpl.yaml (hunk 1 above)
Comment (magorKathy) [reply charts/gateway-helm/values.tmpl.yaml]: fixed. thanks

**Comment by zirain** [comment]
please fix DCO, and run `make -k gen-check`

**Comment by zirain** [comment]
donot merge main, this one is simple.
Otherwise I need approve CI for first-time contributor again and again, thanks!

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
thanks !

**Pull request merged**